reward/staking_manager.go

```diff
@@ -161,6 +177,11 @@ func updateStakingInfo(blockNum uint64) (*StakingInfo, error) {
 		return nil, err
 	}
 
+	// Fill in Gini coeff before adding to cache and DB.
+	if err := fillMissingGiniCoefficient(stakingInfo, blockNum); err != nil {
+		logger.Warn("Cannot fill in gini coefficient", "blockNum", blockNum, "err", err)
+	}
+
```

Comment:
This could update existing DB's stakingInfo due to the following `AddStakingInfoToDB()`, resulting in non-consistency among multiple nodes. Is this ok?

Reply:
1. Before this PR, DB always stores `{Gini: -1}`.
2. After this PR, DB sometimes stores `{Gini: 0.xx}`.
    ```
    //   If cache hit     -> fillMissingGini -> modifies cached in-memory object
    //   If db hit        -> fillMissingGini -> write to cache
    //   If read contract -> fillMissingGini -> write to db -> write to cache
    ```
    In this case, DB content may differ among nodes depending on when they have upgraded to v1.9.0.

How about I change the fixup logic like below @ian0371?

```diff
  //   If cache hit     -> fillMissingGini -> modifies cached in-memory object
  //   If db hit        -> fillMissingGini -> write to cache
- //   If read contract -> fillMissingGini -> write to db -> write to cache
+ //   If read contract -> write to db (-1) -> fillMissingGini -> write to cache (0.xx)
```

Reply:
LGTM :)

Reply:
Fixed in 4032a013.